client/src/views/Models/components/DrilldownPanel/MetadataPane.vue

```diff
@@ -78,27 +115,65 @@
 <script lang="ts">
   import Vue from 'vue';
   import Component from 'vue-class-component';
-  import { Prop } from 'vue-property-decorator';
+  import { Prop, Watch } from 'vue-property-decorator';
   import _ from 'lodash';
 
+  import { emmaaEntityInfo, emmaaEvidence } from '@/services/EmmaaFetchService';
+
+  import { truncateString } from '@/utils/StringUtil';
+
   import * as GroMET from '@/types/typesGroMEt';
+  import * as EMMAA from '@/types/typesEmmaa';
+
   import { formatFullDateTime } from '@/utils/DateTimeUtil';
   import MessageDisplay from '@/components/widgets/MessageDisplay.vue';
+  import LoadingAlert from '@/components/widgets/LoadingAlert.vue';
 
   const METADATA_TYPES_ORDER = [
     GroMET.MetadataType.TextDefinition,
     GroMET.MetadataType.TextParameter,
     GroMET.MetadataType.EquationDefinition,
     GroMET.MetadataType.CodeSpanReference,
+    GroMET.MetadataType.CodeSpanReference,
+    GroMET.MetadataType.IndraAgentReferenceSet,
+    GroMET.MetadataType.ReactionReference,
   ];
 
   const components = {
     MessageDisplay,
+    LoadingAlert,
   };
 
   @Component({ components })
   export default class MetadataPane extends Vue {
     @Prop({ default: [] }) metadata: GroMET.Metadata[];
+    @Prop({ default: '' }) modelName: string;
+
+    isLoading = false;
+    agentsReferences: EMMAA.EmmaaEntityInfoInterface[] = [];
+    nodeEvidences: EMMAA.EmmaaEvidenceEvidenceInterface[] = [];
+
+    @Watch('metadata') onDataChange (): void {
+      if (this.metadata.find(datum => datum.metadata_type === GroMET.MetadataType.IndraAgentReferenceSet)) {
```

Comment:
You can simplify your tests inside `onDataChange ()` and `mounted ()` by using the new methods you wrote `isTypeIndraAgentReferenceSet` and `isTypeReactionReference`.
That would keep the `metadata_type` test in one place in case we have to update them.

```JS
// Fom:
this.metadata.find(datum => datum.metadata_type === GroMET.MetadataType.IndraAgentReferenceSet)

// To:
this.metadata.find(this.isTypeIndraAgentReferenceSet)
```

Reply:
Great point, thanks!